src/core_modules/capture-core/metaDataMemoryStoreBuilders/programs/factory/enrollment/EnrollmentFactory.js

```diff
@@ -91,63 +85,138 @@ export class EnrollmentFactory {
+    ) {
         const section = new Section((o) => {
             o.id = Section.MAIN_SECTION_ID;
             o.name = i18n.t('Profile');
         });
 
-        featureTypeField && section.addElement(featureTypeField);
-        if (cachedProgramTrackedEntityAttributes && cachedProgramTrackedEntityAttributes.length > 0) {
-            // $FlowFixMe
-            await cachedProgramTrackedEntityAttributes.asyncForEach(async (ptea) => {
-                const element = await this.dataElementFactory.build(ptea);
-                element && section.addElement(element);
-            });
+        if (!cachedProgramTrackedEntityAttributes?.length) { return null; }
+
+        if (cachedProgramTrackedEntityTypeId) {
+            const featureTypeField = this._buildTetFeatureTypeField(cachedProgramTrackedEntityTypeId);
+            featureTypeField && section.addElement(featureTypeField);
+        }
+
+        await this._buildElementsForSection(cachedProgramTrackedEntityAttributes, section);
+        return section;
+    }
+
+    async _buildElementsForSection(
+        cachedProgramTrackedEntityAttributes: ?Array<CachedProgramTrackedEntityAttribute>,
+        section: Section,
+    ) {
+        // $FlowFixMe
+        await cachedProgramTrackedEntityAttributes.asyncForEach(async (trackedEntityAttribute) => {
+            const element = await this.dataElementFactory.build(trackedEntityAttribute);
+            element && section.addElement(element);
+        });
+        return section;
+    }
+
+    async _buildSection(
+        cachedProgramTrackedEntityAttributes?: Array<CachedProgramTrackedEntityAttribute>,
+        cachedSectionCustomLabel: string,
+        cachedSectionCustomId: string,
+    ) {
+        if (!cachedProgramTrackedEntityAttributes?.length) {
+            return null;
         }
 
+        const section = new Section((o) => {
+            o.id = cachedSectionCustomId;
+            o.name = cachedSectionCustomLabel;
+        });
+
+        await this._buildElementsForSection(cachedProgramTrackedEntityAttributes, section);
         return section;
     }
 
+    async _buildCustomEnrollmentForm(
+        enrollmentForm: RenderFoundation,
+        dataEntryForm: CachedDataEntryForm,
+        cachedProgramTrackedEntityAttributes: ?Array<CachedProgramTrackedEntityAttribute>,
+    ) {
+        if (!cachedProgramTrackedEntityAttributes) { return null; }
+
+        let section = new Section((o) => {
+            o.id = Section.MAIN_SECTION_ID;
+        });
+
+        section.showContainer = false;
+
+        section = await this._buildElementsForSection(cachedProgramTrackedEntityAttributes, section);
+        section && enrollmentForm.addSection(section);
+        try {
+            enrollmentForm.customForm = new CustomForm((o) => {
+                o.id = dataEntryForm.id;
+            });
+            // $FlowFixMe : Require input from class
+            enrollmentForm.customForm.setData(dataEntryForm.htmlCode, transformTrackerNode);
```

Comment:
NIT: Flow declaration part 2

Reply:
I think I've been more specific on the flow declarations, but having some trouble on what to change on this one. The transformTrackerNode takes node and index as input and they are both sent in from the `CustomForm`-class. Therefore, this callback-function will not have any input even though flow expects there to be. Could you give me a hint of what to do? Also be sure to let me know if you still think the other declarations could be improved. Thanks!

Reply:
Made a suggestion for this (see other comment)